src/twisted/trial/test/__init__.py

```diff
@@ -4,3 +4,39 @@
 """
 Unit tests for the Trial unit-testing framework.
 """
+
+from hypothesis import HealthCheck, settings
+
+
+def _activateHypothesisProfile():
+    """
+    Load a Hypothesis profile appropriate for a Twisted test suite.
+    """
+    deterministic = settings(
+        # Disable the deadline.  It is too hard to guarantee that a particular
+        # piece of Python code will always run in less than some fixed amount
+        # of time.  Hardware capabilities, the OS scheduler, the Python
+        # garbage collector, and other factors all combine to make substantial
+        # outliers possible.  Such failures are a distraction from development
+        # and a hassle on continuous integration environments.
+        deadline=None,
+        suppress_health_check=[
+            # With the same reasoning as above, disable the Hypothesis time
+            # limit on data generation by example search strategies.
+            HealthCheck.too_slow,
+        ],
+        # When a developer is working on one set of changes, or continuous
+        # integration system is testing them, it is disruptive for Hypothesis
+        # to discover a bug in pre-existing code.  This is just what
+        # Hypothesis will do by default, by exploring a pseudo-randomly
+        # different set of examples each time.  Such failures are a
+        # distraction from development and a hassle in continuous integration
+        # environments.
+        derandomize=True,
```

Comment:
I guess that later we can optionally enable random data injected via env vars?

In this way, we can enable a nightly trunk CI run with random runs.

But right now, I don't think that we have the time to look after nightly CI test results

Reply:
Enable random mode somehow, yea.  I filed https://github.com/twisted/twisted/issues/11671 because I think first-class support for profile selection in trial might be a good thing.  In other projects I have just used environment variables so if #11671 doesn't look appealing or is too much work I know we can always fall back to that approach.

I agree that it's not clear right now who would be responsible for _doing_ anything with the nightly random results so it doesn't seem to make sense to run them yet - but hopefully we'll get there.